script.js

```diff
@@ -1,4 +1,238 @@
+    p2DicePosition = dicePosition;
+    // List the dice values using p1DiceValues with the helper function
+    // Match it to Player 2
+    getNumOfDiceAndRoll(p1DiceValues.length);
+    p2DiceValues = diceValues;
+    gameMode = 'Player 2 Enter Position';
+    // Output Player 2 dice values and prompt index question.
+    myOutputValue = 'Player 2, you have decided to <br>play with ' + numOfDice + ' dice. <br><br>You rolled: <br>---- dice: ' + p2DiceValues + '<br>position: ' + p2DicePosition + '<br><br>Enter the positions of the numbers you want.';
+    return myOutputValue;
+  }
+
+  if (gameMode == 'Player 2 Enter Position') {
+    // Check if input is a number and is not an empty string.
+    if (isNaN(Number(input)) == false && input != '' && input.length == p2DiceValues.length) {
+      // Put input into helper function to generate player's score
+      generatePlayerScore(input);
+      player2Score = Number(playerScore);
+      totalp2Score += player2Score;
+      myOutputValue = 'Player 2, your score is: ' + player2Score;
+      // On click, auto compute the total scores to determine the winner.
+      gameMode = 'Compare Scores';
+    }
+    else {
+      myOutputValue = invalidMessage;
+    }
+    // Empty the global variable of helper function output to create a new value for player 1.
+    diceValues = [];
+    dicePosition = [];
+    return myOutputValue;
+  }
+  // Determine the winner by comparing scores.
+  if (gameMode == 'Compare Scores') {
+    if (totalp1Score > totalp2Score) {
+      myOutputValue = 'Player 1 score: ' + totalp1Score + '<br>Player 2 score: ' + totalp2Score + '. <br><br>Player 1 wins! Key in "normal" or "variable" to restart game.';
+    }
+    else if (totalp1Score < totalp2Score) {
+      myOutputValue = 'Player 1 score: ' + totalp1Score + '<br>Player 2 score: ' + totalp2Score + '. <br><br>Player 2 wins! Key in "normal" or "variable" to restart game.';
+    }
+    else {
+      myOutputValue = 'It is a draw! Key in "normal" or "variable" to restart game.';
+    }
+    // Change game mode to the start before returning it.
+    gameMode = 'Enter Mode';
+    return myOutputValue;
+  }
+
+  // Start Normal Dice Game
+  // On Click, player 1 rolls 2 dice.
+  if (gameMode == 'normal') {
+    gameMode = 'Player 1 Choose';
+    myOutputValue = 'Welcome Player 1. <br>You rolled Dice 1: ' + p1DiceOne + ' and Dice 2: ' + p1DiceTwo + '.<br>Choose the order of the dice.';
+    return myOutputValue;
+  }
+  // On Click, player 1 chooses which dice goes first.
+  if (gameMode == 'Player 1 Choose') {
+    gameMode = 'Player 2 Rolls';
+    // But first, check if input is 1 or 2.
+    if (!(input == 1 || input == 2)) {
+      gameMode = 'Player 1 Choose';
+      myOutputValue = 'Please enter 1 or 2 only.';
+      return myOutputValue;
+    }
+    // If input is 1 or 2, the numbers will be concatenated to form a 2 digit number.
+    if (input == 1) {
+      player1CombinedNum = p1DiceOne * 10 + p1DiceTwo;
```

Comment:
This logic can be abstracted to a helper function to avoid repeating yourself on lines 196, 199, 217, and 220. Repeating logic more easily results in bugs